packages/ensnode-sdk/src/ens/metadata-service.ts

```diff
@@ -0,0 +1,273 @@
+      parsed.assetName.namespace !== "erc721" &&
+      parsed.assetName.namespace !== "erc1155"
+    ) {
+      return false;
+    }
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Function type for generating browser-supported asset URLs through a custom proxy.
+ *
+ * @param name - The ENS name to get the browser supported asset URL for
+ * @param assetUrl - The asset URL parsed as a URL object, allowing protocol-specific logic (e.g., ipfs:// vs ar://)
+ * @param namespaceId - The ENS namespace identifier for the name
+ * @returns The browser supported asset URL, or null if unavailable
+ */
+export type BrowserSupportedAssetUrlProxy = (
+  name: Name,
+  assetUrl: URL,
+  namespaceId: ENSNamespaceId,
+) => BrowserSupportedAssetUrl | null;
+
+/**
+ * Default proxy implementation that uses the ENS Metadata Service.
+ *
+ * @param name - The ENS name to get the browser supported asset URL for
+ * @param assetUrl - The asset URL (not used in default implementation)
+ * @param namespaceId - The ENS namespace identifier for the name
+ * @returns The browser supported asset URL from ENS Metadata Service, or null if unavailable
+ */
+export const defaultBrowserSupportedAssetUrlProxy: BrowserSupportedAssetUrlProxy = (
+  name: Name,
+  assetUrl: URL,
+  namespaceId: ENSNamespaceId,
+): BrowserSupportedAssetUrl | null => {
+  return buildEnsMetadataServiceAvatarUrl(name, namespaceId);
+};
+
+/**
+ * Result returned by buildBrowserSupportedAvatarUrl.
+ *
+ * Invariant: If rawAssetTextRecord is null, then browserSupportedAssetUrl must also be null.
+ */
+export interface BrowserSupportedAssetUrlResult {
+  /**
+   * The original asset text record value from ENS, before any normalization or proxy processing.
+   * Null if the asset text record is not set for the ENS name.
+   */
+  rawAssetTextRecord: string | null;
+  /**
+   * A browser-supported (http/https/data) asset URL ready for use in <img> tags or other browser contexts.
+   * Populated when the rawAssetTextRecord is a valid URL that uses a browser-supported protocol (http, https, or data) or when a url is available to load the asset using a proxy.
+   * Null if the asset text record is not set, if the asset text record is malformed/invalid,
+   * or if the asset uses a non-browser-supported protocol and no url is known for how to load the asset using a proxy.
+   */
+  browserSupportedAssetUrl: BrowserSupportedAssetUrl | null;
+  /**
+   * Indicates whether the browserSupportedAssetUrl uses a proxy service.
+   * True if the url uses a proxy (either the default ENS Metadata Service or a custom proxy).
+   * False if the URL comes directly from the asset text record, or if there's no asset text record,
+   * or if the asset text record has an invalid format, or if no url is known for loading the asset using a proxy.
+   */
+  usesProxy: boolean;
+}
+
+/**
+ * Builds a browser-supported asset URL for a name's asset image from the name's raw asset text record value.
```

Comment:
```suggestion
* Builds a browser‑supported avatar image URL from the ENS avatar text record (key: "avatar").
* Accepts non‑browser protocols (ipfs://, ar://) and NFT URIs (CAIP‑22 ERC‑721 / CAIP‑29 ERC‑1155, eip155 chain namespace),
* and converts them via a proxy/gateway (default: ENS Metadata Service) to http/https/data URLs.
```

This keeps the "asset" generic type for reuse, but adjusts the JSDoc to emphasize the use of the "avatar text record" in this context.